recipe/migrations/abseil_cpp202206230.yaml

```
@@ -0,0 +1,20 @@
+__migrator:
+  build_number: 1
+  kind: version
+  migration_number: 2
+
+# TODO: Make sure to remove the associated zip keys in the final file
+# when the migration is complete
+abseil_cpp:
+- '20220623.0'
+# new name, use both for now
+libabseil:
+- '20220623.0'
+# separate output for static builds (depending on C++ version)
+# also need to be migrated
+libabseil_static:
+- '20220623.0'
+grpc_cpp:
+- '1.48'
```

grpc builds have been merged. I don't see another reason for the zip_keys between abseil & grpc, except historical difficulty & convenience.

Still, obviously 1.46-1.48 work fine with the new abseil, so I think it'd be good to take out the zip_keys both from the migratior and the global pinning. That way, feedstocks can choose their own speed with updating grpc (not least since 1.47+ now requires at least C++14, and this might be necessary to propagate to consuming feedstocks)

---

There are some peculiarity with migrating. I'm worried that without the zip, something will fail. We can take it out once it has all been merged. IMO

---

I don't feel very strongly about this (so fine either way) but not sure it helps TBH. If we merge the end of this migration we'll be at 1.48 globally, and downgrading is even more of a hassle.

Any feedstock with problems could still try any grpc between 1.46 - 1.48.

But maybe the golden middle is just to zip this with 1.46, so we're not forcing all feedstocks to make (as big a jump) in grpc:
```suggestion
- '1.46'
```